docs/Dialects/FIRRTL/RationaleFIRRTL.md

```diff
@@ -567,21 +567,31 @@ include the following:
 
 - Ports on a `firrtl.module`, where the foreign types are treated as opaque values moving in and out of the module
 - Ports on a `firrtl.instance`
-- `firrtl.wire` to allow for def-after-use cases; the wire must have a single strict connect that uniquely defines the wire's value
-- `firrtl.matchingconnect` to module outputs, instance inputs, and wires
+- `firrtl.foreign_output` 
+
+Foreign types on ports are set with the connect-like operation 
+`firrtl.foreign_output`.  This may be the only use of a foreign-typed output 
```

Comment:
This greatly simplifies LowerToHW, since it has to transform the ports and there isn't a great way to partially transform that boundary.  One might be able to use a dialect-trait, but then one might want to add traits to built-in dialects.  Having a single output op makes the transformation simple.

Reply:
Mainly, guaranteeing a single output-port connect operation for each foreign-type port makes conversion from connect to hw.output style trivial.  Without this, one might need to add wires, which we no longer could do.

Reply:
Okay, yeah for LowerToHW.  Makes sense, okay thanks!